src/features/forecastapi/forecast-api.ts

```diff
@@ -13,17 +13,23 @@ namespace ForecastApiUtilities {
 
   /**
    * Gets non project time types from forecast
-   * 
+   *
    * @returns All non project times where is_internal_time is false
    */
   export const getNonProjectTime = async (): Promise<NonProjectTime[]> => {
     try {
       const result = await fetch(`${process.env.forecast_base_url}non_project_time`, { headers: headers });
-      const NonProjectTimes: NonProjectTime[] = await result.json();
-      return NonProjectTimes.filter(NonProjectTime => NonProjectTime.is_internal_time !== true);
+      const nonProjectTimes = await result.json();
+      if (nonProjectTimes[0].id){
+        const filterRes: NonProjectTime[] = nonProjectTimes.filter(nonPTime => nonPTime.is_internal_time !== true);
+        if (filterRes.length) {
+          return filterRes;
+        }
+      }
+      throw new Error(`Error while loading non project time, ${nonProjectTimes.message}`);
     } catch(error) {
-      console.error("Error while loading non project time");
-      Promise.reject(error);
+      console.error(error);
+      return Promise.reject(error);
```

Comment:
No need to return promise reject. Promise.reject() will automatically return out from the current function
```suggestion
      Promise.reject(error);
```

Reply:
if this is not returned we get an UnhandledPromiseRejection and several tests fail as, investigating further...

Reply:
done